Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/LogExceptionAnalyzer.cs

```diff
@@ -0,0 +1,106 @@
+﻿// © 2020 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability
+{
+	/// <summary>
+	/// Report when a catch exception block does not call one of the logging methods.
+	/// </summary>
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class LogExceptionAnalyzer : DiagnosticAnalyzer
+	{
+		private const string Title = "Log caught exceptions.";
+		private const string Message = "Exception that is caught, is not logged.";
+		private const string Description = "Log caught exceptions.";
+		private const string Category = Categories.Maintainability;
+
+		private static readonly DiagnosticDescriptor Rule =
+			new DiagnosticDescriptor(
+				Helper.ToDiagnosticId(DiagnosticIds.LogException),
+				Title,
+				Message,
+				Category,
+				DiagnosticSeverity.Error,
+				isEnabledByDefault: false,
+				description: Description
+			);
+
+		private string logMethodNames;
+
+		/// <summary>
+		/// <inheritdoc/>
+		/// </summary>
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		/// <summary>
+		/// <inheritdoc/>
+		/// </summary>
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction(
+				startContext =>
+				{
+					var additionalFiles = new AdditionalFilesHelper(
+						startContext.Options,
+						startContext.Compilation);
+					var methodNames = additionalFiles.GetValueFromEditorConfig(Rule.Id, "log_method_names");
+					if (string.IsNullOrEmpty(methodNames))
+					{
+						startContext.RegisterCompilationEndAction(ReportParsingError);
+					}
+					else
+					{
+						logMethodNames = methodNames;
+						context.RegisterSyntaxNodeAction(AnalyzeCatchException, SyntaxKind.CatchClause);
```

Comment:
This should say startContext, which will get rid of the above warning.

Reply:
Indeed, done